File: 1184 - Distance Between Bus Stops.py
class Solution:
    def distanceBetweenBusStops(self, distance: list[int], start: int, destination: int) -> int:
        pathA, pathB = 0, 0
        res = 0
        a = min(start, destination)
        b = max(start, destination)
        for i in range(a, b):
            pathA += distance[i]

        i = b
        while i != a:
            pathB += distance[i]
            i = (i + 1) % len(distance)
        res = min(pathA, pathB)
        return res

File: 1184 - Distance Between Bus Stops.js
/**
 * @param {number[]} distance
 * @param {number} start
 * @param {number} destination
 * @return {number}
 */
var distanceBetweenBusStops = function (distance, start, destination) {
    let pa = 0, pb = 0;
    let a = Math.min(start, destination);
    let b = Math.max(start, destination);

    for (let i = a; i < b; i++) {
        pa += distance[i];

    }
    for (let i of distance) {
        pb += i;
    }
    pb -= pa;
    return Math.min(pa, pb);
};